OPHD/UI/Core/RadioButton.cpp

```diff
@@ -33,60 +26,57 @@ RadioButton::~RadioButton()
  * \note	Setting check state with this method will not fire
  *			a clicked event.
  */
-void RadioButton::checked(bool toggle)
+void RadioButtonGroup::RadioButton::checked(bool toggle)
 {
 	mChecked = toggle;
 }
 
 
-bool RadioButton::checked() const
+bool RadioButtonGroup::RadioButton::checked() const
 {
 	return mChecked;
 }
 
-void RadioButton::parentContainer(UIContainer* parent)
+void RadioButtonGroup::RadioButton::parentContainer(RadioButtonGroup* parent)
 {
 	mParentContainer = parent;
 }
 
-void RadioButton::text(const std::string& text)
+void RadioButtonGroup::RadioButton::text(const std::string& text)
 {
 	mLabel.text(text);
 	onTextChange();
 }
 
-const std::string& RadioButton::text() const
+const std::string& RadioButtonGroup::RadioButton::text() const
 {
 	return mLabel.text();
 }
 
-RadioButton::ClickSignal::Source& RadioButton::click()
+
+RadioButtonGroup::RadioButton::ClickCallback& RadioButtonGroup::RadioButton::click()
 {
-	for (auto* sibling : mParentContainer->controls())
-	{
-		if (auto* asRadioButton = dynamic_cast<RadioButton*>(sibling))
-		{
-			asRadioButton->checked(false);
-		}
-	}
+	mParentContainer->clearSelection();
+	mRbgDelegate();
```

Comment:
I kind of feel like `clearSelection` should be handled internally in the `mRbgDelegate()` callback method. That way `clearSelection` could potentially be made `private` or `protected`.

----

This is the wrong function to put this code in. This method is returning the `SignalSource` object. It is not the `onClick` handler method.

Reply:
Ok, I'm now treating the method that returns the SignalSource just as a getter for the SignalSource (meaning removing any state altering logic and only return the object. Is that the correct interpretation?

Reply:
I think there is still some confusion here, I noticed the mCallback (which is different from the mRbgDelegate) was in fact not used, I've removed mCallback and its getter/setter to make the situation clearer.

Member `mRbgDelegate` contains the function that is executed in the targeted UI Control (here: FactoryProduction window `chkIdleClicked`).

Although as an inner class of RadioButtonGroup I don't really see the problem with `mParentContainer` calling `clearSelection()`.

Reply:
Sounds like `mRbgDelegate` is being used for what I assumed `mSignal` was being used for. Might be a good idea to rename the field. Having "RBG" in the name makes me assume the callback is from the `RadioButton` to the `RadioButtonGroup` object, so it can implement one-hot. I expected `mSignal` to be for the final user method that's called when a corresponding option is selected.

Looks like there is no longer any accessor method for the `SignalSource` interface.

Reply:
okay, renamed field mRbgDelegate to mSignal.